b_cfn_opensearch_index/source/main/action.py

```diff
@@ -0,0 +1,216 @@
+                try:
+                    self.opensearch_client.reindex(
+                        body={
+                            'source': {
+                                'index': index_name
+                            },
+                            'dest': {
+                                'index': new_index_name
+                            }
+                        },
+                        refresh=True,
+                        timeout='10m'
+                    )
+                except Exception as ex:
+                    logger.warning(f'Failed to reindex documents. Reason: {repr(ex)}.')
+
+            return ActionResult(
+                data={'IndexName': new_index_name},
+                resource_id=f'{self.opensearch_domain_endpoint}||{new_index_name}'
+            )
+        else:
+            # Check does requested index settings update holds any of static index settings.
+            # If yes raise an exception because static index settings can be set only at the
+            # time of index creation, but cannot be changed/updated.
+            if any(key in self.index_settings for key in static_index_settings):
+                raise Exception(
+                    f'Failed to update settings of the live index. '
+                    f'Reason: Static index settings cannot be updated. {static_index_settings=}'
+                )
+
+            try:
+                # Update dynamic index settings.
+                self.opensearch_client.indices.put_settings(
+                    index=index_name,
+                    body=self.index_settings,
+                    timeout='5m'
+                )
+            except Exception as ex:
+                raise Exception(f'Failed to update settings of the live index. Reason: {repr(ex)}.')
+
+            try:
+                # Update index mappings.
+                self.opensearch_client.indices.put_mapping(
+                    index=index_name,
+                    body=self.index_mapping_settings,
+                    timeout='5m'
+                )
+            except Exception as ex:
+                raise Exception(f'Failed to update mappings of the live index. Reason: {repr(ex)}.')
+
+            return ActionResult(data={'IndexName': index_name}, resource_id=self.__resource_id)
+
+    def delete(self) -> ActionResult:
+        """
+        Deletes an Opensearch index with index name which is parsed from PhysicalResourceId.
+
+        :return: Delete action result.
+        """
+
+        logger.info(f'Initiating resource delete with these properties: {json.dumps(self.__properties)}.')
+
+        index_name = self.__resource_id.split('||')[1]
+
+        self.opensearch_client.indices.delete(
+            index=index_name,
+            expand_wildcards='all'
+        )
+        return ActionResult(data={'IndexName': index_name}, resource_id=self.__resource_id)
+
+    @staticmethod
+    def validate_index_prefix(index_prefix: str) -> None:
```

Comment:
Is this even necessary? Opensearch will raise an error anyways if the name is not compliant. Why bother with this functionality??